drivers/sensor/sngcja5/sngcja5.c

```diff
@@ -0,0 +1,179 @@
+{
+	uint8_t buf[2];
+	int ret;
+
+	ret = read_bytes(dev, addr, &buf[0], 2);
+	if (ret) {
+		LOG_ERR("read_bytes() @ i2c failed");
+		return -EIO;
+	}
+
+	*value = (buf[1] << 8) | buf[0];
+
+	return 0;
+}
+
+static int sngcja5_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	struct sngcja5_data *device_data = dev->data;
+	const struct sngcja5_config *device_config = dev->config;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL);
+
+	/* Read out all relevant sensor value registers */
+	if (read_register_4(&device_config->i2c, SNGCJA5_PM10_LL, &(device_data->pm1_0))) {
+		return -EIO;
+	}
+	if (read_register_4(&device_config->i2c, SNGCJA5_PM25_LL, &(device_data->pm2_5))) {
+		return -EIO;
+	}
+	if (read_register_4(&device_config->i2c, SNGCJA5_PM100_LL, &(device_data->pm10_0))) {
+		return -EIO;
+	}
+	if (read_register_2(&device_config->i2c, SNGCJA5_05_L, &(device_data->pc0_5))) {
+		return -EIO;
+	}
+	if (read_register_2(&device_config->i2c, SNGCJA5_10_L, &(device_data->pc1_0))) {
+		return -EIO;
+	}
+	if (read_register_2(&device_config->i2c, SNGCJA5_25_L, &(device_data->pc2_5))) {
+		return -EIO;
+	}
+	if (read_register_2(&device_config->i2c, SNGCJA5_50_L, &(device_data->pc5_0))) {
+		return -EIO;
+	}
+	if (read_register_2(&device_config->i2c, SNGCJA5_75_L, &(device_data->pc7_5))) {
+		return -EIO;
+	}
+	if (read_register_2(&device_config->i2c, SNGCJA5_100_L, &(device_data->pc10_0))) {
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int sngcja5_channel_get(const struct device *dev, enum sensor_channel chan,
+			       struct sensor_value *val)
+{
+	struct sngcja5_data *device_data = dev->data;
+	int32_t uval;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL);
+	if (chan != SENSOR_CHAN_ALL) {
+		return -ENOTSUP;
+	}
+
+	val[0].val1 = device_data->pc0_5;
+	val[0].val2 = 0;
+	val[1].val1 = device_data->pc1_0;
+	val[1].val2 = 0;
+	val[2].val1 = device_data->pc2_5;
+	val[2].val2 = 0;
+	val[3].val1 = device_data->pc5_0;
+	val[3].val2 = 0;
+	val[4].val1 = device_data->pc7_5;
+	val[4].val2 = 0;
+	val[5].val1 = device_data->pc10_0;
+	val[5].val2 = 0;
+
+	uval = (int32_t)device_data->pm1_0;
+	val[6].val1 = uval / SNGCJA5_SCALE_FACTOR;
+	val[6].val2 = uval % SNGCJA5_SCALE_FACTOR;
+	uval = (int32_t)device_data->pm2_5;
+	val[7].val1 = uval / SNGCJA5_SCALE_FACTOR;
+	val[7].val2 = uval % SNGCJA5_SCALE_FACTOR;
+	uval = (int32_t)device_data->pm10_0;
+	val[8].val1 = uval / SNGCJA5_SCALE_FACTOR;
+	val[8].val2 = uval % SNGCJA5_SCALE_FACTOR;
+
+	return 0;
+}
+
+int sngcja5_init(const struct device *dev)
+{
+	const struct sngcja5_config *config = dev->config;
+
+	if (!device_is_ready(config->i2c.bus)) {
+		LOG_ERR("I2C bus device is not ready");
+		return -ENODEV;
+	}
+
+	if (i2c_configure(config->i2c.bus,
+			  I2C_SPEED_SET(I2C_SPEED_STANDARD) | I2C_MODE_CONTROLLER)) {
+		LOG_ERR("i2c_configure() failed");
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static const struct sensor_driver_api sngcja5_driver_api = {
```

Comment:
I'm always getting following error message in the twister runs: 
`zephyr/drivers/sensor/sngcja5/sngcja5.c:163:39: error: 'sngcja5_driver_api' defined but not used [-Werror=unused-const-variable=]`

Can someone explain me what I'm doing wrong, or what I have to do to pass?

Reply:
There probably aren't any enabled instances of `panasonic,sngcja5` in your devicetree

Reply:
But I've added the _compatible = "panasonic,sngcja5";_ to my _arduino_i2c.overlay_ file. So i thought that it has to be in the devicetree afterwards?

Reply:
is `status = okay` ? do you see the node in the build artifact `zephyr.dts`? if you don't, that would suggest you're not including the overlay in the build

Reply:
Yes I see the node in the build artifact zephyr.dts. Don't getting this error message in the twister runs anymore.